02_bubble/sahidakhtar1/main.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	list := []int{3, 2, 1, 5}
+	bubbleSort(list)
+	fmt.Fprintln(out, "After sorting = ", list)
+}
+
+func bubbleSort(list []int) {
+	count := len(list)
+	for count > 0 {
+		swappedIndex := 0
+		for index := 1; index < count; index++ {
+			if list[index-1] > list[index] {
+				list[index-1], list[index] = list[index], list[index-1]
+				swappedIndex = index
+			}
+		}
+		count = swappedIndex
+	}
+}
```

Comment:
Can't mutate the original. Make a copy and return it. Also write a test to check that the original isn't mutated.

Reply:
That means make a `copy()` of list and modify it.

Reply:
`bubbleSort ()` will return new list without modifying the input. Also added Unit test case to check input is not modified `TestBubbleSortInput`.